src/App.tsx

```diff
@@ -1,25 +1,56 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const total = 42;
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const items = getNumbers(1, total);
+  const totalPages = Math.ceil(items.length / perPage);
+
+  const indexOfLastItem = currentPage * perPage;
+  const indexOfFirstItem = indexOfLastItem - perPage;
+  const currentItems = items.slice(indexOfFirstItem, indexOfLastItem);
+
+  function handleSelectChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    setPerPage(Number(event.target.value));
+    setCurrentPage(1);
```

Comment:
When changing the 'perPage' value, it's a good practice to also check if the current page is still valid. For instance, if you're on the last page and you increase the number of items per page, the current page might no longer exist. Adjusting the current page to the last valid page in such cases would improve user experience.

Reply:
I don't understand what you mean. Why do I need to check the current page if every time we change perPage current page becomes 1, it is part of the condititons of the task. Moreover, there is only valid values in select(3, 5, 10, 20)